### PR TITLE
add checkstyle validation, fix errors

### PR DESCRIPTION
**Pull request by zml2008**
Starting out a checkstyle file for some basic formatting things.

There are several custom checks used -- see https://gitlab.com/stellardrift/stylecheck for their source. For now, i've put in a check that will prohibit TYPE_USE nullability annotations, but not perform any further validation.

~~This also doesn't verify usage of nullability annotations or prohibiting TYPE_USE annotations on local variables. I'm going to explore writing an annotation processor to do that.~~

### CODE REVIEW

File: api/src/main/java/net/kyori/adventure/text/NBTComponentImpl.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public int hashCode() {
     );
   }
 
-  static abstract class BuilderImpl<C extends NBTComponent<C, B>, B extends NBTComponentBuilder<C, B>> extends AbstractComponentBuilder<C, B> implements NBTComponentBuilder<C, B> {
+  abstract static class BuilderImpl<C extends NBTComponent<C, B>, B extends NBTComponentBuilder<C, B>> extends AbstractComponentBuilder<C, B> implements NBTComponentBuilder<C, B> {
```
Comment (kashike) [inline api/src/main/java/net/kyori/adventure/text/NBTComponentImpl.java:82]: Is there a way to change the checkstyle configuration to keep this `static abstract`?

File: etc/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  <property name="severity" value="error"/>
+  <property name="fileExtensions" value="java, properties, xml"/>
+  <!-- Excludes all 'module-info.java' files              -->
+  <!-- See https://checkstyle.org/config_filefilters.html -->
+  <module name="BeforeExecutionExclusionFileFilter">
+    <property name="fileNamePattern" value="module\-info\.java$"/>
+  </module>
+  <!-- https://checkstyle.org/config_filters.html#SuppressionFilter -->
+  <module name="SuppressionFilter">
+    <property name="file" value="${basedir}/checkstyle/checkstyle-suppressions.xml"
```
Comment (kashike) [inline etc/checkstyle/checkstyle.xml:16]: Assuming `${basedir}` is the repository root, then `${basedir}/.checkstyle/suppressions.xml`

File: api/src/main/java/net/kyori/adventure/text/NBTComponentImpl.java (hunk 1 above)
Comment (zml2008) [reply api/src/main/java/net/kyori/adventure/text/NBTComponentImpl.java:82]: Sure -- this is done by the `ModifierOrder` checker, which I can just take out of the config.

If you have a specific preferred order, let me know and I can toss that in to the extra checkers library.

File: etc/checkstyle/checkstyle.xml (hunk 2 above)
Comment (zml2008) [reply etc/checkstyle/checkstyle.xml:16]: oh oops, I got this wrong, basedir is `<project root>/etc/checkstyle`, where the `checkstyle.xml` currentl is. Want to go for both the `checkstyle.xml` and `suppressions.xml` in `<project root>/.checkstyle`?

Comment (kashike) [reply etc/checkstyle/checkstyle.xml:16]: both in there works good for me

**Comment by zml2008** [comment]
Alight, I think I've reached a point where I've covered everything I want to do in this round of style checks.

@kashike if there's anything that bugs you still let me know. If not, I'll apply styling to the other projects, push out a release of `stylecheck`, and then merge this in.

(the build seems to have failed -- i suspect because the latest `stylecheck` snapshot has not made it through sonatype OSS and into travis's cache)

File: text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializer.java [hunk 3]
```diff
@@ -82,7 +82,7 @@ static Builder builder() {
   /**
    * The legacy character used by Minecraft. ('§')
    */
-  char SECTION_CHAR = '\u00A7';
+  char SECTION_CHAR = '§';
```
[the remark this answers is not in the document]
Comment (zml2008) [reply text-serializer-legacy/src/main/java/net/kyori/adventure/text/serializer/legacy/LegacyComponentSerializer.java:85]: was aiming to match the JD comment above which uses the character literal -- and hey, we have UTF-8 source files :p

It's based on a checkstyle rule to [prefer writing characters directly over writing them as unicode escapes](https://github.com/KyoriPowered/adventure/pull/86/files#diff-300096062b5a39fc537e1fc611e0ff05R131)

**Pull request merged**